exprs.go

```diff
@@ -538,11 +557,11 @@ func (up *unboundUnaryPredicate) Bind(schema *Schema, caseSensitive bool) (Boole
 	// fast case optimizations
 	switch up.op {
 	case OpIsNull:
-		if bound.Ref().Field().Required {
+		if bound.Ref().Field().Required && !schema.FieldHasOptionalParent(bound.Ref().Field().ID) {
```

Comment:
not sure what you mean. Write a value at an index when the parent value at the same index is null (at least in Parquet that would be invalid), or do you mean are you asking whether it's possible for a required field to have an optional parent? (Yup, it's possible. It means that when the parent is non-null, this field must also be non-null, otherwise there's no value in this field at that index)